### PR TITLE
Reorganize the book.

### PR DESCRIPTION
**Pull request by hdevalence**
This PR has one unfortunate change, which is that the README.md and
CONTRIBUTING.md files in the book are symlinks to files in the parent
directory.  The motivation for this is to ensure that we don't maintain two
copies of the same data, and that the landing page of the website matches the
landing page of the Github repo, etc.   However, I'm not sure whether these
symlinks will work correctly on Windows.

The alternatives are:

- Duplicate the contents of the files and expect that people will know to keep
  them in sync;

- Use relative links `../../README.md` in the `SUMMARY.md`.  This seemed like
  it caused mdbook to dump the rendered files into the repository root rather
  than keeping them in the `book` directory.

- Use a symlink (chosen option).  This may not work on Windows but I think that
  the worst outcome would be that the book would be unbuildable unless someone
  used WSL or something.  This seems like the least bad option.

### CODE REVIEW

**Comment by yaahc** [comment]
> However, I'm not sure whether these symlinks will work correctly on Windows.

They almost certainly won't, I ran into this same problem when I was implementing the zcashconsensus rust crate. I tried symlinking the header file from the git submodule, windows hated it.

If there is a way to do an `include` where we source the markdown from another directory into the current file that would be my goto solution. We do something like that in `awesome-rust-mentors`, but that is leveraging include templating in jekyll, so we'd have to see if theres an mdbook equivalent. https://github.com/RustBeginners/awesome-rust-mentors#project-mentorship

**Comment by hdevalence** [comment]
OK, I'm happy to go with option 1 (duplicate the files) if people would prefer that.  I'd prioritize getting something everyone is happy with and merging these changes over the choice of which option to do.  I'm not sure if mdbook can do includes but I'll check it out.

**Comment by yaahc** [comment]
@hdevalence looks like theres a `links` plugin for mdbook that we can enable to handle this

https://github.com/rust-lang/mdBook#3rd-party-plugins

**Comment by hdevalence** [comment]
Looks like includes may be possible: https://github.com/rust-lang/mdBook/issues/245

**Comment by hdevalence** [comment]
@yaahc Thanks, fixed!!

**Review by dconnolly — APPROVED** [review]
I love Book 📚

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Contributing
```
Comment (dconnolly) [inline CONTRIBUTING.md:1]: 🎉

**Pull request merged**